pkl-executor/src/main/java/org/pkl/executor/ExecutorOptions2.java

```diff
@@ -62,7 +64,8 @@ public ExecutorOptions2(
       /* @Nullable */ Path moduleCacheDir,
       /* @Nullable */ Path projectDir,
       List<Path> certificateFiles,
-      List<URI> certificateUris) {
+      List<URI> certificateUris,
+      int testPort) {
```

Comment:
Since `testPort` is an internal test-only option, we should have another constructor that doesn't have it as an option, and sets `testPort` to `-1`.

Reply:
Should probably address the root cause and provide a builder for `ExecutorOptions(2)`. For example, almost nobody will want to pass certificates either.

I thought about modernizing the `pkl-executor` API to look more like `Evaluator` and language bindings. But then I realized that my time is better spent on Java language bindings, which have the potential to make `pkl-executor` unnecessary.

Reply:
Added a builder. See commit message for details.